src/sqlitedb.cpp

```diff
@@ -225,13 +228,59 @@ bool DBBrowserDB::tryEncryptionSettings(const QString& filename, bool* encrypted
         {
             sqlite3_finalize(vm);
 #ifdef ENABLE_SQLCIPHER
+            if (!isDatabasePasswordChecked)
+            {
+                isDatabasePasswordChecked = true;
+
+                QMap<QString, QVariant> databasePasswords = PreferencesDialog::getSettingsValue("db", "databasepasswords").toMap();
+
+                if (databasePasswords.count() > 0)
+                {
+                    QFile databaseFile(filePath);
+                    QFileInfo databaseFileInfo(databaseFile);
+                    QString databaseFileName(databaseFileInfo.baseName());
```

Comment:
This is a bit confusing because it omits the file extension: 'mydbfile.db' bcomes 'mydbfile' because of the baseName() function. So, for adding a saved password for my 'mydbfile.db' file I'd have to enter 'mydbfile' here without the extension. And to be honest, I would have never guessed that without first looking at the code. So I'd replace the baseName() function by a fileName() call.